src/runtime/virtcontainers/remote.go

```diff
@@ -0,0 +1,288 @@
+
+	return nil
+}
+
+func (rh *remoteHypervisor) AttestVM(ctx context.Context) error {
+	return nil
+}
+
+func (rh *remoteHypervisor) StopVM(ctx context.Context, waitOnly bool) error {
+
+	s, err := openRemoteService(rh.config.RemoteHypervisorSocket)
+	if err != nil {
+		return err
+	}
+	defer s.Close()
+
+	req := &pb.StopVMRequest{
+		Id: string(rh.sandboxID),
+	}
+
+	if _, err := s.client.StopVM(ctx, req); err != nil {
+		return fmt.Errorf("remote hypervisor call failed: %w", err)
+	}
+
+	return nil
+}
+
+func (rh *remoteHypervisor) GenerateSocket(id string) (interface{}, error) {
+
+	socketPath := rh.agentSocketPath
+	if len(socketPath) == 0 {
+		return nil, errors.New("failed to generate remote sock: TunnelSocketPath is not set")
+	}
+
+	remoteSock := types.RemoteSock{
+		SandboxID:        id,
+		TunnelSocketPath: socketPath,
+	}
+
+	return remoteSock, nil
+}
+
+func notImplemented(name string) error {
+
+	err := errors.Errorf("%s: not implemented", name)
+
+	logrus.Errorf(err.Error())
+
+	if tracer, ok := err.(interface{ StackTrace() errors.StackTrace }); ok {
+		for _, f := range tracer.StackTrace() {
+			logrus.Errorf("%+s:%d\n", f, f)
+		}
+	}
+
+	return err
+}
+
+func (rh *remoteHypervisor) PauseVM(ctx context.Context) error {
+	return notImplemented("PauseVM")
+}
+
+func (rh *remoteHypervisor) SaveVM() error {
+	return notImplemented("SaveVM")
+}
+
+func (rh *remoteHypervisor) ResumeVM(ctx context.Context) error {
+	return notImplemented("ResumeVM")
+}
+
+func (rh *remoteHypervisor) AddDevice(ctx context.Context, devInfo interface{}, devType DeviceType) error {
+	// TODO should we return notImplemented("AddDevice"), rather than nil and ignoring it?
+	logrus.Printf("addDevice: deviceType=%v devInfo=%#v", devType, devInfo)
+	return nil
+}
+
+func (rh *remoteHypervisor) HotplugAddDevice(ctx context.Context, devInfo interface{}, devType DeviceType) (interface{}, error) {
+	return nil, notImplemented("HotplugAddDevice")
+}
+
+func (rh *remoteHypervisor) HotplugRemoveDevice(ctx context.Context, devInfo interface{}, devType DeviceType) (interface{}, error) {
+	return nil, notImplemented("HotplugRemoveDevice")
+}
+
+func (rh *remoteHypervisor) ResizeMemory(ctx context.Context, memMB uint32, memoryBlockSizeMB uint32, probe bool) (uint32, MemoryDevice, error) {
+	return memMB, MemoryDevice{}, nil
+}
+
+func (rh *remoteHypervisor) GetTotalMemoryMB(ctx context.Context) uint32 {
+	return rh.config.MemorySize
+}
+
+func (rh *remoteHypervisor) ResizeVCPUs(ctx context.Context, vcpus uint32) (uint32, uint32, error) {
+	return vcpus, vcpus, nil
+}
+
+func (rh *remoteHypervisor) GetVMConsole(ctx context.Context, sandboxID string) (string, string, error) {
+	return "", "", notImplemented("GetVMConsole")
+}
+
+func (rh *remoteHypervisor) Disconnect(ctx context.Context) {
+	notImplemented("Disconnect")
+}
+
+func (rh *remoteHypervisor) Capabilities(ctx context.Context) types.Capabilities {
+	var caps types.Capabilities
+	caps.SetBlockDeviceHotplugSupport()
```

Comment:
Is it true for peer pods? i.e. does it support block device hotplug? Also wondering whether `Capabilities` should be part of hypervisor protocol or not.

Reply:
@stevenhorsman @bpradipt do you see any use of `Capabilities` in the protocol?

Reply:
@wainersm I think we'll need Capabilities in the protocol.  But not sure if BlockDeviceHotplugSupport is used somewhere.
@yoheiueda @huoqifeng do you recall if there is a need for SetBlockDeviceHotplugSupport ?

Reply:
I think to support CSI in remoteHyperVisor, BlockDeviceHotplugSupport is needed but I'll deffer it to @yoheiueda to confirm that.

Reply:
Yes. it has been introduced to support CSI in peer pods.

I think the capability was set when I was working on this problem with @genjuro214 https://github.com/kata-containers/kata-containers/pull/3697/commits/c3e6b660515046fef704e60d36efb05c10b5060b